modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -168,6 +169,65 @@ export class Wallets implements IWallets {
     };
   }
 
+  private async generateLightningWallet(params: GenerateWalletOptions): Promise<LightningWalletWithKeychains> {
+    const reqId = new RequestTracer();
+    this.bitgo.setRequestTracer(reqId);
+
+    const lightningParams = _.pick(params, ['label', 'passphrase', 'enterprise', 'passcodeEncryptionCode']);
+    const { label, passphrase, enterprise, passcodeEncryptionCode } = lightningParams;
+
+    assert(label && typeof label === 'string', 'label parameter is required and must be a string');
+    assert(passphrase && typeof passphrase === 'string', 'passphrase parameter is required and must be a string');
+    assert(enterprise && typeof enterprise === 'string', 'enterprise parameter is required and must be a string');
+    assert(
+      passcodeEncryptionCode && typeof passcodeEncryptionCode === 'string',
+      'passcodeEncryptionCode parameter is required and must be a string'
+    );
+
+    const keychainPromises = ([undefined, 'userAuth', 'nodeAuth'] as const).map((purpose) => {
+      return async (): Promise<Keychain> => {
+        const keychain = this.baseCoin.keychains().create();
+        const keychainParams: AddKeychainOptions = {
+          pub: keychain.pub,
+          encryptedPrv: this.bitgo.encrypt({ password: passphrase, input: keychain.prv }),
+          originalPasscodeEncryptionCode: purpose === undefined ? passcodeEncryptionCode : undefined,
+          coinSpecific: purpose === undefined ? undefined : { [this.baseCoin.getChain()]: { purpose } },
+          keyType: 'independent',
+          source: 'user',
+        };
+        const newKeychain = await this.baseCoin.keychains().add(keychainParams);
+        return _.extend({}, newKeychain, keychain);
+      };
+    });
+
+    const { userKeychain, userAuthKeychain, nodeAuthKeychain } = await promiseProps({
+      userKeychain: keychainPromises[0](),
+      userAuthKeychain: keychainPromises[1](),
+      nodeAuthKeychain: keychainPromises[2](),
+    });
+
+    const walletParams: SupplementGenerateWalletOptions = {
+      label,
+      m: 1,
+      n: 1,
+      type: 'hot',
+      enterprise,
+      keys: [userKeychain.id],
```

Comment:
For lightning, what is the userKeychain used for?

Reply:
Also have you run this function against staging/test? 

Im surprised wallet platform works with just 1 keyId as input instead of 3.

Reply:
We recently added support for single user key wallet support for lightning in WP.

Reply:
> Also have you run this function against staging/test?
yes, worked